File: models/task.go
package models

import "time"

type Task struct {
	Id          int    `json:"id" gorm:"type:int; primaryKey; autoIncrement"`
	UserId      int    `json:"userId" gorm:"type:int"`
	Title       string `json:"title" gorm:"type:varchar(255)"`
	Description string `json:"description" gorm:"type:text"`
	Status      string `json:"status" gorm:"type:varchar(50)"`
	// default nya kosong jadi jika tidak ada isinya maka diisi "" (kosong)
	Reason       string    `json:"reason" gorm:"type:text; default:"`
	Revision     int       `json:"revision" gorm:"type:int; default:0"`
	DueDate      string    `json:"dueDate" gorm:"type:varchar(50)"`
	SubmitDate   string    `json:"submitDate" gorm:"type:varchar(50)"`
	RejectedDate string    `json:"rejectedDate" gorm:"type:varchar(50)"`
	ApprovedDate string    `json:"approvedDate" gorm:"type:varchar(50)"`
	Attachment   string    `json:"attachment" gorm:"type:varchar(255)"`
	CreatedAt    time.Time `json:"createdAt" gorm:"type:timestamp; default:CURRENT_TIMESTAMP"`
	UpdatedAt    time.Time `json:"updatedAt" gorm:"type:timestamp; default:NULL"`
	DeletedAt    time.Time `json:"deletedAt" gorm:"type:timestamp; default:NULL"`
	User         User      `json:"user,omitempty" gorm:"foreignKey:UserId"` // Punya :
}

File: tusk_app/lib/data/models/task.dart
import 'package:tusk_app/data/models/user.dart';

class Task {
  final int? id;
  final int? userId;
  final String? title;
  final String? description;
  final String? status;
  final String? reason;
  final int? revision;
  final DateTime? dueDate;
  final DateTime? submitDate;
  final DateTime? rejectedDate;
  final DateTime? approvedDate;
  final String? attachment;
  final DateTime? createdAt;
  final DateTime? updatedAt;
  final User? user;

  Task({
    this.id,
    this.userId,
    this.title,
    this.description,
    this.status,
    this.reason,
    this.revision,
    this.dueDate,
    this.submitDate,
    this.rejectedDate,
    this.approvedDate,
    this.attachment,
    this.createdAt,
    this.updatedAt,
    this.user,
  });

  factory Task.fromJson(Map<String, dynamic> json) => Task(
        id: json["id"],
        userId: json["userId"],
        title: json["title"],
        description: json["description"],
        status: json["status"],
        reason: json["reason"],
        revision: json["revision"],
        dueDate:
            json["dueDate"] == null ? null : DateTime.parse(json["dueDate"]),
        submitDate: json["submitDate"] == null || json["submitDate"] == ''
            ? null
            : DateTime.parse(json["submitDate"]),
        rejectedDate: json["rejectedDate"] == null || json["rejectedDate"] == ''
            ? null
            : DateTime.parse(json["rejectedDate"]),
        approvedDate: json["approvedDate"] == null || json["approvedDate"] == ''
            ? null
            : DateTime.parse(json["approvedDate"]),
        attachment: json["attachment"],
        createdAt: json["createdAt"] == null
            ? null
            : DateTime.parse(json["createdAt"]),
        updatedAt: json["updatedAt"] == null
            ? null
            : DateTime.parse(json["updatedAt"]),
        user: json["user"] == null ? null : User.fromJson(json["user"]),
      );

  Map<String, dynamic> toJson() => {
        "id": id,
        "userId": userId,
        "title": title,
        "description": description,
        "status": status,
        "reason": reason,
        "revision": revision,
        "dueDate": dueDate?.toIso8601String(),
        "submitDate": submitDate?.toIso8601String(),
        "rejectedDate": rejectedDate?.toIso8601String(),
        "approvedDate": approvedDate?.toIso8601String(),
        "attachment": attachment,
        "createdAt": createdAt?.toIso8601String(),
        "updatedAt": updatedAt?.toIso8601String(),
        "user": user?.toJson(),
      };
}
